### PR TITLE
[CORE] depracate get/set_concatenation_axis functions

### PR DESCRIPTION
**Pull request by barnasm1**
### Details:
 - remove `set_concatenation_axis` function usage
 - replace `get_concatenation_axis` function usage

### Tickets:
- [CVS-94507](https://jira.devtools.intel.com/browse/CVS-94507)

### Related PR:
 - https://github.com/openvinotoolkit/openvino/pull/24383

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/concatenation.cpp [hunk 1]
```diff
@@ -69,7 +69,6 @@ std::vector<layout> concatenation_inst::calc_output_layouts(const concatenation_
     }
     ov::op::v0::Concat op;
     op.set_friendly_name(desc->id);
-    op.set_concatenation_axis(axis_index);
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/concatenation.cpp:72]: Why did you remove that? I think it should be replaced with `set_axis()` instead

Comment (barnasm1) [reply src/plugins/intel_gpu/src/graph/concatenation.cpp:72]: Done

File: src/common/low_precision_transformations/src/move_fake_quantize.cpp [hunk 2]
```diff
@@ -83,7 +84,11 @@ bool MoveFakeQuantize::transform(TransformationContext& context, ov::pass::patte
     if (concat_node == nullptr) {
         return false;
     }
-    const auto concat_axis = concat_node->get_concatenation_axis();
+    int64_t concat_axis = -1;
```
Comment (jane-intel) [inline src/common/low_precision_transformations/src/move_fake_quantize.cpp:87]: why -1 and not the concat_node->get_axis()?

File: src/core/include/openvino/op/concat.hpp [hunk 3]
```diff
@@ -37,9 +37,11 @@ class OPENVINO_API Concat : public Op {
     std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override;
 
     /// \return The concatenation axis.
+    OPENVINO_DEPRECATED("This method will be removed in 2025.0")
     int64_t get_concatenation_axis() const {
         return m_concat_axis;
     }
+    OPENVINO_DEPRECATED("This method will be removed in 2025.0")
```
Comment (jane-intel) [inline src/core/include/openvino/op/concat.hpp:44]: lets encourage people to use get_axis/set_axis methods

File: src/common/transformations/src/transformations/transpose_sinking/ts_concat.cpp [hunk 4]
```diff
@@ -81,11 +83,13 @@ TSConcatBackward::TSConcatBackward() {
...
 
         auto concat_node = as_type_ptr<ov::op::v0::Concat>(main_node);
-        auto concat_axis = concat_node->get_concatenation_axis();
-        if (concat_axis < 0) {
+        if (concat_node->get_output_partial_shape(0).is_dynamic()) {
             return false;
         }
 
+        const auto rank = concat_node->get_output_partial_shape(0).rank().get_length();
+        auto concat_axis = ov::util::normalize(concat_node->get_axis(), rank);
```
[the remark this answers is not in the document]
Comment (barnasm1) [reply src/common/transformations/src/transformations/transpose_sinking/ts_concat.cpp:91]: Done

File: src/common/offline_transformations/src/pruning/propagate_masks.cpp [hunk 5]
```diff
@@ -765,7 +765,11 @@ class ov::pass::mask_propagation::Concat : public MatcherPass {
             if (!concat_ptr) {
                 return false;
             }
-            auto axis = concat_ptr->get_concatenation_axis();
+            auto axis = -1;
+            if (concat_ptr->get_output_partial_shape(0).rank().is_static()) {
+                const auto rank = concat_ptr->get_output_partial_shape(0).rank().get_length();
+                axis = ov::util::normalize(concat_ptr->get_axis(), rank);
+            }
```
[the remark this answers is not in the document]
Comment (barnasm1) [reply src/common/offline_transformations/src/pruning/propagate_masks.cpp]: What is the decision? Should I replace current solution with if-else approach?

File: src/common/low_precision_transformations/src/move_fake_quantize.cpp (hunk 2 above)
Comment (barnasm1) [reply src/common/low_precision_transformations/src/move_fake_quantize.cpp:87]: Done

Comment (barnasm1) [reply src/common/low_precision_transformations/src/move_fake_quantize.cpp:87]: My understanding is that function `get_concatenation_axis()` returns -1 if rank is dynamic. So current code mimic previous implementation. If we use `get_axis()` directly the behavior of the function will change or it will require additional check to assign -1 in case of dynamic rank. Is it correct or I missed something?

File: src/core/include/openvino/op/concat.hpp (hunk 3 above)
Comment (barnasm1) [reply src/core/include/openvino/op/concat.hpp:44]: Done

**Review by praasz — APPROVED** [review]

File: src/common/low_precision_transformations/src/move_fake_quantize.cpp (hunk 2 above)
Comment (barnasm1) [reply src/common/low_precision_transformations/src/move_fake_quantize.cpp]: With support of analysis by @praasz the suggestion was applied.

**Comment by mlukasze** [comment]
Hey @vladimir-paramuzov  & @itikhono 
Could you please review or delegate someone to review GPU & Transformations parts, please?

**Review by vladimir-paramuzov — APPROVED** [review]
GPU part LGTM
